File: wheel/src/main/java/dev/yong/wheel/utils/ToastUtils.java
package dev.yong.wheel.utils;

import android.os.Looper;
import android.view.Gravity;
import android.widget.Toast;

import dev.yong.wheel.AppManager;

/**
 * Toast工具类
 *
 * @author CoderYong
 */

public class ToastUtils {

    private ToastUtils() {
        throw new UnsupportedOperationException("Cannot be created");
    }

    private static long sTime = -1;

    public static int GRAVITY = Gravity.NO_GRAVITY;

    /**
     * Toast提示
     *
     * @param text     提示文本
     * @param duration 显示时长
     */
    public static void show(CharSequence text, int duration) {
        long intervals = 2000;
        if (System.currentTimeMillis() - sTime > intervals) {
            try {
                Toast toast = Toast.makeText(AppManager.getInstance().getApplication(), text, duration);
                toast.setGravity(GRAVITY, 0, 0);
                toast.show();
            } catch (Exception e) {
                //解决在子线程中调用Toast的异常情况处理
                Looper.prepare();
                Toast toast = Toast.makeText(AppManager.getInstance().getApplication(), text, duration);
                toast.setGravity(GRAVITY, 0, 0);
                toast.show();
                Looper.loop();
            }
            sTime = System.currentTimeMillis();
        }
    }

    /**
     * Toast提示
     *
     * @param text 提示文本 (默认短时间显示)
     */
    public static void show(CharSequence text) {
        show(text, Toast.LENGTH_SHORT);
    }

    /**
     * Toast提示
     *
     * @param resId    提示文本资源Id
     * @param duration 显示时长
     */
    public static void show(int resId, int duration) {
        show(AppManager.getInstance().getApplication().getString(resId), duration);
    }

    /**
     * Toast提示 (默认短时间显示)
     *
     * @param resId 提示文本资源Id
     */
    public static void show(int resId) {
        show(resId, Toast.LENGTH_SHORT);
    }
}

File: wheel/src/main/java/dev/yong/wheel/utils/ToastUtils.kt
package dev.yong.wheel.utils

import android.content.Context
import android.view.Gravity
import android.view.View
import android.widget.Toast
import androidx.annotation.StringRes
import androidx.fragment.app.Fragment

/**
 * Toast显示位置
 * @see android.widget.Toast.setGravity
 */
var GRAVITY = Gravity.BOTTOM
var X_OFFSET = 0
var Y_OFFSET = 0

/**
 * Toast间距
 * @see android.widget.Toast.setMargin
 */
var HORIZONTAL_MARGIN = 0F
var VERTICAL_MARGIN = 0F

/**
 * Toast 提示
 *
 * @param message  提示内容
 * @param duration 提示时长
 */
fun Context.toast(message: CharSequence, duration: Int = Toast.LENGTH_SHORT) {
    val toast = Toast.makeText(this, message, duration)
    toast.setGravity(GRAVITY, X_OFFSET, Y_OFFSET)
    toast.setMargin(HORIZONTAL_MARGIN, VERTICAL_MARGIN)
    toast.show()
}

/**
 * Toast 提示
 *
 * @param resId    提示字符串资源ID
 * @param duration 提示时长
 */
fun Context.toast(@StringRes resId: Int, duration: Int = Toast.LENGTH_SHORT) {
    toast(getString(resId), duration)
}

/**
 * Toast 提示
 *
 * @param view     自定义提示View
 * @param duration 提示时长
 */
@Suppress("DEPRECATION")
fun Context.toast(view: View, duration: Int = Toast.LENGTH_SHORT) {
    val toast = Toast(this)
    toast.view = view
    toast.duration = duration
    toast.setGravity(GRAVITY, X_OFFSET, Y_OFFSET)
    toast.setMargin(HORIZONTAL_MARGIN, VERTICAL_MARGIN)
    toast.show()
}


/**
 * Toast 提示
 *
 * @param message  提示内容
 * @param duration 提示时长
 */
fun Fragment.toast(message: CharSequence, duration: Int = Toast.LENGTH_SHORT) {
    requireContext().toast(message, duration)
}

/**
 * Toast 提示
 *
 * @param resId    提示字符串资源ID
 * @param duration 提示时长
 */
fun Fragment.toast(@StringRes resId: Int, duration: Int = Toast.LENGTH_SHORT) {
    requireContext().toast(resId, duration)
}

/**
 * Toast 提示
 *
 * @param view     自定义提示View
 * @param duration 提示时长
 */
fun Fragment.toast(view: View, duration: Int = Toast.LENGTH_SHORT) {
    requireContext().toast(view, duration)
}